osu.Game/Beatmaps/Drawables/DifficultyIconTooltip.cs

```diff
@@ -70,8 +153,50 @@ public void SetContent(DifficultyIconTooltipContent content)
 
             displayedContent = content;
 
+            // Header row
             starRating.Current.BindTarget = displayedContent.Difficulty;
             difficultyName.Text = displayedContent.BeatmapInfo.DifficultyName;
+
+            // Don't show difficulty stats if showExtendedTooltip is false
+            if (!displayedContent.ShowExtendedTooltip) return;
+
+            // Show the difficulty stats if showExtendedTooltip is true
+            difficultyFillFlowContainer.Show();
+            miscFillFlowContainer.Show();
```

Comment:
The alpha on the elements are set to 0 by default. This seems a bit redundant when they're already hidden. The alpha value should be set to 1 only if showExtendedTooltip is true.

Reply:
Yes, but `SetContent` can be called *multiple times*. The whole premise behind these tooltips being structured as they are is that we can update the content via `SetContent` without recreating the full tooltip, allowing less re-initialisation but also animating between different content.

So imagine a sequence:

`SetContent(expanded:true)`
`SetContent(expanded:false)`

Please add test coverage to allow testing this and you'll see what I mean immediately.

Reply:
Ah I get it now. Will definitely add tests to be able to visualize this better.

Reply:
Applied the suggested changes. The tests show how the expanded content continues to be visible despite ShowExpandedTooltip being false. Thanks for pointing it out!

I've also added testing coverage for DifficultyIcon. Not sure If I placed it in the right directory, but I can easily move it somewhere else. If the location of the test isn't an issue, this can be marked as resolved.